src/ripple/app/ledger/tests/common_ledger.cpp

```diff
@@ -432,29 +432,67 @@ trust(TestAccount& from, TestAccount const& issuer,
 void
 close_and_advance(Ledger::pointer& ledger, Ledger::pointer& LCL)
 {
+    // This version of close_and_advance uses wall time.
+    using namespace std::chrono;
+    auto const epoch_offset = days(10957);  // 2000-01-01
+    auto const closeTime = time_point_cast<seconds>  // now
+        (system_clock::now() - epoch_offset).
+        time_since_epoch().count();
+
+    // Pass the time delta to the other close_and_advance.
+    close_and_advance (
+        ledger, LCL, closeTime - ledger->getParentCloseTimeNC());
+}
+
+void
+close_and_advance(
+    Ledger::pointer& ledger, Ledger::pointer& LCL, std::uint32_t advSeconds)
+{
+    // An empty container of transactions to retry.
+    auto retries = LocalTxs::New();
+
+    // Let the most complicated version of close_and_advance do the lifting.
+    close_and_advance (ledger, LCL, advSeconds, retries);
+}
+
+void
+close_and_advance(Ledger::pointer& ledger, Ledger::pointer& LCL,
+    std::uint32_t advSeconds, std::unique_ptr<LocalTxs>& retries)
+{
     std::shared_ptr<SHAMap> set = ledger->peekTransactionMap();
     CanonicalTXSet retriableTransactions(set->getHash());
     Ledger::pointer newLCL = std::make_shared<Ledger>(false, *LCL);
     // Set up to write SHAMap changes to our database,
     //   perform updates, extract changes
-    applyTransactions(set, newLCL, newLCL, retriableTransactions, false);
+    {
+        TransactionEngine engine (newLCL, tx_enable_test);
+        applyTransactions(set, engine, newLCL, retriableTransactions, false);
+    }
     newLCL->updateSkipList();
     newLCL->setClosed();
     newLCL->peekAccountStateMap()->flushDirty(
         hotACCOUNT_NODE, newLCL->getLedgerSeq());
```

Comment:
That's a fair question, and I don't know the answer.  `close_and_advance` has had a few hands in its creation.  The part you are referring to was present when I got here.  I've studied the function well enough to modify it, but not enough to truly understand it.